src/java/org/apache/cassandra/index/sai/disk/v1/SegmentMetadata.java

```diff
@@ -69,8 +69,8 @@ public class SegmentMetadata implements Comparable<SegmentMetadata>
     /**
      * Ordered by their token position in current segment
      */
-    public final PrimaryKey minKey;
-    public final PrimaryKey maxKey;
+    public final PrimaryKey minPartitionKey;
```

Comment:
Is it a partition key or a primary key?  Does this have the clustering columns in it?  If so it is a PrimaryKey not a PartitionKey.

Reply:
We always create it as a partition key:

https://github.com/datastax/cassandra/blob/8dd873ec34a76f8e08ff30a2776ca5848142277f/src/java/org/apache/cassandra/index/sai/disk/v1/MemtableIndexWriter.java#L179-L187

Reply:
Sure, I'll add one.

In general, I think this gets to a point that @blambov was making in a different github ticket. The fact that equality can mean different things depending on how much of your key is present creates a confusing mental model and leads to bugs like this one, where the assumption was that the list has unique PrimaryKeys, therefore Collections.binarySearch will provide a valid result.

Reply:
agreed.